src/nvim/file_search.c

```diff
@@ -1522,6 +1523,35 @@ find_file_in_path_option (
   return file_name;
 }
 
+static void apply_autocmd_dirchanged(char_u *new_dir, CdScope scope)
+{
+  dict_T *dict = get_vim_var_dict(VV_EVENT);
+  char buf[8];
```

Comment:
A check like the one in `yank_do_autocmd` would be worth adding here.

```
    if (recursive || !has_event(EVENT_DIRCHANGED)) {
      // No autocommand was defined 
      // or we changed directory from this autocommand. 
      return;
    }
```

Speaking of which, let's disable recursive DirChanged. (We can always add it later if it is needed for some reason)